src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -793,3 +912,165 @@ g.test('empty_bind_group_layouts_requires_empty_bind_groups,render_pass')
       encoder.finish();
     }, !success);
   });
+
+g.test('default_bind_group_layouts_never_match,compute_pass')
+  .desc(
+    `
+  Test that bind groups created with default bind group layouts never match other layouts, including empty bind groups.
+
+  * Test that a pipeline with an explicit layout can not be used with a bindGroup from an auto layout
+  * Test that a pipeline with an auto layout can not be used with a bindGroup from an explicit layout
+  * Test that an auto layout from one pipeline can not be used with an auto layout from a different pipeline.
+
+  TODO:
+  * Test empty bindgroup layouts on the same default layout pipeline are not compatible. In other words if
+    you only define group(2) then group(0)'s empty layout and group(1)'s empty layout should be incompatible.
+  `
+  )
+  .params(u =>
+    u
+      .combine('pipelineAndBinding', [
+        { pipelineType: 'auto0', bindingType: 'auto0' }, // successful case
+        { pipelineType: 'explicit', bindingType: 'explicit' }, // successful case
+        { pipelineType: 'explicit', bindingType: 'auto0' },
+        { pipelineType: 'auto0', bindingType: 'explicit' },
+        { pipelineType: 'auto0', bindingType: 'auto1' },
+      ] as const)
+      .combine('empty', [false, true])
```

Comment:
nit: factor this part out into a variable (using kUnitCaseParamsBuilder) to show it's the same across the render and compute test